### PR TITLE
Migrate DatePicker to CSS Modules.

### LINKED ISSUE #551: Convert Datepicker component to CSS modules
# Issue Description
To better align with modern CSS practices and modern CSS theming capabilities, we want to update this component to use CSS modules and if applicable, CSS custom properties.

## Issue Type

- [ ] New Feature
- [x] Enhancement
- [ ] Bug
- [ ] Other

## Expected Behavior
Component styles are converted to use CSS modules. If applicable, convert SCSS variables to CSS custom properties.

## Current Behavior
Component styles use manual name spaced classes and SCSS variables.

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Closes #551. 

Note: I did not convert date-rage to css modules since the implementation is not currently in use and there is no css styling applied.

@cerner/terra

### CODE REVIEW

File: packages/terra-date-picker/CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ Changelog
 
 Unreleased
 ----------
-
+* Converted component to use CSS modules
```
Comment (benbcai) [inline packages/terra-date-picker/CHANGELOG.md]: Add '### Changed'

**Comment by mjhenkes** [comment]
@emilyrohrbough could you resolve the conflicts on this branch?

**Comment by PaulSolDev** [comment]
Should the date format MM/DD/YYYY auto format dates entered manually to include the forward slashes ' / '?

**Comment by emilyrohrbough** [comment]
@benbcai Do you have the answer to Paul's question?

**Comment by benbcai** [comment]
@PaulSolDev The DatePicker currently does not support any text entry masking. It is just a free text input. The entry is accepted/validated only when entered correctly according to the format. the There is an enhancement (https://github.com/cerner/terra-core/issues/303) to enforce formatting during entry.

**Comment by PaulSolDev** [comment]
@benbcai Thank you for the clarification.  So long as there is something logged then we should be good.  I will comment back again when we have completed the remaining browser testing.

**Comment by PaulSolDev** [comment]
Functional review complete.  No issues identified outside of the need to include date formatting in the future.

**Pull request merged**